Sources/swift-bootstrap/main.swift

```diff
@@ -330,6 +334,9 @@ struct SwiftBootstrapBuildTool: ParsableCommand {
                     fileSystem: self.fileSystem,
                     observabilityScope: self.observabilityScope
                 )
+                #else
+                fatalError("SwiftPM was built without XCBuild support")
```

Comment:
This line is technically unreachable, since `case .xcode` is never reachable on non-Darwin platforms, and we don't call `swift-bootstrap --build-system xcode` anywhere during the build process.

Reply:
> This line is technically unreachable

Heh, famous last words:
https://ci.swift.org/job/oss-swift-package-macos/2753

```
--- bootstrap: note: Building SwiftPM (with a freshly built swift-bootstrap)
main/main.swift:345: Fatal error: SwiftPM was built without XCBuild support
```

Reply:
```
bootstrap/bin/swift-bootstrap ... --arch x86_64 --arch arm64 ...
```

And:
```
    private var buildSystem: BuildSystemProvider.Kind {
        #if os(macOS)
        // Force the Xcode build system if we want to build more than one arch.
        return self.architectures.count > 1 ? .xcode : .native
        #else
        // Force building with the native build system on other platforms than macOS.
        return .native
        #endif
    }
```